website/src/firebase-utils/login.jsx

```diff
@@ -28,23 +29,28 @@ function promptForLogin() {
   });
 }
 
-/** 
- * Manages the login for a user, if they are not logged in prompt for a log in.
- */
-export function manageLogin() {
-  firebase.auth().onAuthStateChanged(user => {
-    if (user) {
-      uidCallback(user.uid);
-    } else {
-      promptForLogin();
-    }
-  });
-}
-
-/** 
+/**
  * Signs the user out and redirects them to the landing page
+ * @returns {void}
  */
 export function signOut() {
   firebase.auth().signOut();
   document.location = "/#/login";
 }
+
+/**
+ * Gets user infos and updates the state
+ * @param  {function} dispatch The dispatch function for redux
+ * @return {void}
+ */
+export function getUserInfo(dispatch) {
+  firebase.auth().onAuthStateChanged((user) => {
```

Comment:
this function sets up a callback on the auth state changed event, so any time the auth state changes this will get called (also it should only be called once, otherwise it will get run multiple times for every event). This is why users don't stay logged out, because they sign out and then this this function ends up calling promptForLogin and then the cookies in their browser automatically log them back in